### ISSUE TITLE
datahike.core does not validate transactions against schema

### ISSUE THREAD

**Issue by theronic** [replikativ/datahike#178]
datahike.core does not validate schema when transacting because it seems to follow DataScript semantics, e.g.:

```
(require '[datahike.core :as d])
(d/datoms
    (:db-after (d/with (d/empty-db)
                 [[:db/add -1 :broken/attr 367]]))
    :eavt)
  => (#datahike/Datom[1 :broken/attr 367 536870913 true])
```

This makes it tricky to test. So I see I can use the datahike.connector namespace to validate transactions against schema, but now I have to first define an in-memory configuration, create a database, then connect to it:

```
(require '[datahike.connector :as dc])
(let [config {:store {:backend :mem :id "mem-example"}}]
  (dc/create-database config)
  (let [conn (dc/connect config)]
    @(dc/transact! conn [[:db/add -1 :broken/attr 367]])))
=> Execution error (ExceptionInfo) at datahike.db/validate-attr (db.cljc:986).
No schema found in db.
```

It would be convenient to use the datahike.core namespace with schema flexiblity options (schema on read or write) and have it perform tx validation. I think this is what most users would expect from the core namespace.

This caught me out because I was trying to write tests against datahike.core with bad transaction data did not throw on `d/with`.

**Comment by kordano** [comment]
You're right. We wanted to have this namespace for DataScript compatibility in the beginning. For one of the next releases we wanted to restructure these functions. Alternatively you may directly call `empty-db`from the `db` namespace with the proper config:
``` clojure
  (require '[datahike.core :as c])
  (require '[datahike.db :as db])

  (c/datoms
   (:db-after (c/with (db/empty-db nil {:index :datahike.index/hitchhiker-tree
                                        :keep-history? false
                                        :schema-flexibility :write})
                      [[:db/add -1 :broken/attr 367]]))
   :eavt)
```

**Comment by kordano** [comment]
Fixed with `0.3.1`.
